gluonnlp/data/squad_dataset.py

```diff
@@ -0,0 +1,137 @@
+
+# pylint: disable=
+"""SQuAD dataset."""
+
+__all__ = ['SQuAD']
+
+import json
+import os
+import shutil
+import zipfile
+
+from mxnet.gluon.data import ArrayDataset
+from mxnet.gluon.utils import download, check_sha1, _get_repo_file_url
+from .registry import register
+
+
+@register(segment=['train', 'test'])
+class SQuAD(ArrayDataset):
+    """Stanford Question Answering Dataset (SQuAD) - reading comprehension dataset.
+
+    From
+    https://rajpurkar.github.io/SQuAD-explorer/
+
+    License: CreativeCommons BY-SA 4.0
+
+    Parameters
+    ----------
+    segment : str, default 'train'
+        Dataset segment. Options are 'train' and 'dev'.
+    root : str, default '~/.mxnet/datasets/squad'
+        Path to temp folder for storing data.
+    """
+    def __init__(self, segment='train', root=os.path.join('~', '.mxnet', 'datasets', 'squad')):
+        self._data_file = {'train': ('train-v1.1.zip', 'train-v1.1.json',
+                                     '052a75bf8fdb3e843b8649971658eae8133f9b0e'),
+                           'dev': ('dev-v1.1.zip', 'dev-v1.1.json',
+                                   'e31ad736582b72a8eabd5c0b0a38cb779ed913d7')}
+        root = os.path.expanduser(root)
+
+        if not os.path.isdir(root):
+            os.makedirs(root)
+
+        self._root = root
+        self._segment = segment
+        self._get_data()
+
+        super(SQuAD, self).__init__(self._read_data())
+
+    def _get_data(self):
+        """Load data from the file. Does nothing if data was loaded before
+        """
+        data_archive_name, _, data_hash = self._data_file[self._segment]
+        path = os.path.join(self._root, data_archive_name)
+
+        if not os.path.exists(path) or not check_sha1(path, data_hash):
+            file_path = download(_get_repo_file_url('gluon/dataset/squad', data_archive_name),
+                                 path=self._root, sha1_hash=data_hash)
+
+            with zipfile.ZipFile(file_path, 'r') as zf:
+                for member in zf.namelist():
+                    filename = os.path.basename(member)
+
+                    if filename:
+                        dest = os.path.join(self._root, filename)
+
+                        with zf.open(member) as source, open(dest, 'wb') as target:
+                            shutil.copyfileobj(source, target)
+
+    def _read_data(self):
+        """Read data.json from disk and flats it to the following format:
+        Entry = (question_id, question, context, list_of_answers).
+        Question id and list_of_answers also substituted with indices, so it could be later
+        converted into nd.array
+
+        Returns
+        -------
+        List[Tuple]
+            Flatten list of questions
+        """
+        _, data_file_name, _ = self._data_file[self._segment]
+
+        with open(os.path.join(self._root, data_file_name)) as f:
+            samples = json.load(f)
+
+        return _SQuADJsonParser().get_records(samples)
+
+
+class _SQuADJsonParser:
```

Comment:
It doesn't seem necessary to have a separate class for this. Maybe put `get_records()` as a static method in SQuAD class?

Reply:
use name `_get_records()`